### PR TITLE
add functionality for search bar

### PR DESCRIPTION
**Pull request by giangpham-cfg**
Fixed a bug in the sidebar and added functionality to search for KPI names and circle names

### CODE REVIEW

File: KPI-team1-frontend/src/App.tsx [hunk 1]
```diff
@@ -38,8 +44,7 @@ export default function App() {
     try {
       const { data, error } = await supabase
         .from("circle")
-        .select("circle_name, circle_user!inner(*)")
-        .eq("circle_user.user_id", user.id);
+        .select("circle_name, circle_user!inner(*)");
```
Comment (daczczcz1) [inline KPI-team1-frontend/src/App.tsx:47]: Hmmm, selection of default circles depends on this and it needs to include only circles to which the user is assigned to. Could we create a separate state var for this?

File: KPI-team1-frontend/src/components/SearchResultsList.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  const navigate = useNavigate();
+  return (
+    <div className="bg-gray-100 flex flex-col shadow rounded-md max-h-44 overflow-y-scroll">
+      {results &&
+        results.map((result) => {
+          return (
+            <div
+              className="py-2.5 px-5 hover:bg-gray-200 cursor-pointer"
+              key={result.kpi_id ? result.kpi_id : result.circle_id}
+              onClick={(e) => {
```
Comment (daczczcz1) [inline KPI-team1-frontend/src/components/SearchResultsList.tsx:25]: Should be externalized

File: KPI-team1-frontend/src/components/SearchResultsList.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+                {
+                  result.kpi_id
+                    ? navigate(`/kpi/${result.kpi_id}`)
+                    : navigate(`/kpi/circles/${result.circle_id}`);
+                }
+                setResults([]);
+                setInput("");
+              }}
+            >
+              {result.kpi_name ? result.kpi_name : result.circle_name}
```
Comment (daczczcz1) [inline KPI-team1-frontend/src/components/SearchResultsList.tsx:35]: Should be externalized (this should be a generic component IMO - something like this: https://medium.com/edonec/creating-a-generic-component-with-react-typescript-2c17f8c4386e

File: KPI-team1-frontend/src/components/SearchResultsList.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import { useNavigate } from "react-router-dom";
+import { Kpi } from "../model/kpi";
+
+interface ResultsProps {
+  results: any[];
```
Comment (daczczcz1) [inline KPI-team1-frontend/src/components/SearchResultsList.tsx:5]: Should be generic

File: KPI-team1-frontend/src/pages/LoginPage.tsx [hunk 5]
```diff
@@ -31,7 +31,7 @@ export default function LoginPage(): JSX.Element {
             email: data.user.email,
           };
           setUser(userData); // Update the user state in the App component
-          navigate("/kpi/1"); // todo: will update this route after getting favorite circle (`/kpi/${circles[0]?.circle_user[0]?.circle_id}`)
+          navigate("/kpi/circles/1"); // todo: will update this route after getting favorite circle (`/kpi/${circles[0]?.circle_user[0]?.circle_id}`)
```
Comment (daczczcz1) [inline KPI-team1-frontend/src/pages/LoginPage.tsx:34]: The default circle is already available in user details - up to you to use it

**Review by daczczcz1 — CHANGES_REQUESTED** [review]
One change that I think breaks previous functionality (loading circles), apart from this I think the other ones are kinda important, but there is a lot of important things to do yet, so I if you have a lot to do, I won't insist ;).

File: KPI-team1-frontend/src/App.tsx (hunk 1 above)
Comment (giangpham-cfg) [reply KPI-team1-frontend/src/App.tsx:47]: I reverted to the old code, and this change doesn't affect the other part so everything is ok.

File: KPI-team1-frontend/src/pages/LoginPage.tsx (hunk 5 above)
Comment (giangpham-cfg) [reply KPI-team1-frontend/src/pages/LoginPage.tsx:34]: I'm having trouble getting this particular part to work as expected. Could you please review my code for this section and provide suggestions on how I can fix it? At the moment, it's not functioning properly.

File: KPI-team1-frontend/src/components/Searchbar.tsx [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+      const { data, error } = await supabase.from("circle").select("*");
+      if (error) throw error;
+      const results =
+        data &&
+        data.filter((circle) => {
+          return (
+            value &&
+            circle &&
+            circle.circle_name &&
+            circle.circle_name.toLowerCase().includes(value.toLocaleLowerCase())
```
Comment (daczczcz1) [inline KPI-team1-frontend/src/components/Searchbar.tsx:56]: We could do this on the database side, but for now it's fine

File: KPI-team1-frontend/src/pages/EachKpi.tsx [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+            <div className="text-xl font-medium">{`${renderKpi[0].periodicity
+              .charAt(0)
+              .toUpperCase()}${renderKpi[0].periodicity.slice(1)} KPIs`}</div>
+            <div className="shadow-md border-0 border-primary-light">
+              <DataGrid
+                getRowId={(row) => row.circle_kpidef_id}
+                rows={renderKpi}
+                rowSelection={false}
+                columns={HEADER_KPI_COLUMNS}
+                // onRowClick={(params) => {
```
Comment (daczczcz1) [inline KPI-team1-frontend/src/pages/EachKpi.tsx:143]: To delete

File: KPI-team1-frontend/src/pages/LoginPage.tsx [hunk 8]
```diff
@@ -30,15 +34,20 @@ export default function LoginPage(): JSX.Element {
             id: data.user.id,
             email: data.user.email,
           };
-          setUser(userData); // Update the user state in the App component
-          navigate("/kpi/1"); // todo: will update this route after getting favorite circle (`/kpi/${circles[0]?.circle_user[0]?.circle_id}`)
+          setUser(userData);
+          if (userDetails) {
+            navigate(`/kpi/circles/${userDetails.defaultCircleId}`);
```
Comment (daczczcz1) [inline KPI-team1-frontend/src/pages/LoginPage.tsx:39]: Nice

File: KPI-team1-frontend/src/pages/LoginPage.tsx [hunk 9]
```diff
@@ -30,15 +34,20 @@ export default function LoginPage(): JSX.Element {
...
+            navigate(`/kpi/circles/${circles[0].circle_user[0].circle_id}`);
+          }
         }
         setError("Incorrect email or password. Please retry!");
       }
     } catch (error) {
       console.error(error);
     }
   }
+  console.log("check userDetails", userDetails);
```
Comment (daczczcz1) [inline KPI-team1-frontend/src/pages/LoginPage.tsx:50]: To delete

**Review by daczczcz1 — APPROVED** [review]

File: KPI-team1-frontend/src/components/Sidebar.tsx [hunk 10]
```diff
@@ -56,7 +59,7 @@ export default function Sidebar({
             </NavLink>
             <div>
               <NavLink
-                to={"/kpi/1"}
+                to={"/kpi/circles"}
```
Comment (Aurelieph) [inline KPI-team1-frontend/src/components/Sidebar.tsx:62]: i'm guessing that we still want to have a default "view" if we click on KPIs button 
right now it to shows the uncomplete page 
It this is the goal, we should add a todo to make sure we have something on that page

**Review by Aurelieph — COMMENTED** [review]
Added Small comments but overall this is really good !!

File: KPI-team1-frontend/src/components/Sidebar.tsx (hunk 10 above)
Comment (giangpham-cfg) [reply KPI-team1-frontend/src/components/Sidebar.tsx:61]: yes, we still have a default "view" when click on KPIs button and it was been done by this block of code in KpiPage:
`export default function KpiPage(): JSX.Element {
  const { circleId } = useParams();
  const [selectedCircleId, setSelectedCircleId] = useState<string | null>("");
  const [modalIsOpen, setModalIsOpen] = useState(false);
  const [selectedKpi, setSelectedKpi] = useState<KpiExtended | null>(null);
  const { kpiDefinitions, userDetails }: OutletContext =
    useOutletContext();

  useEffect(() => {
    if (circleId) {
      setSelectedCircleId(circleId);
    } else {
      setSelectedCircleId(userDetails.defaultCircleId);
    }
  }, [circleId]);`

**Pull request merged**